docs_src/examples/use_cases/example_indiv_hvp_retain_graph.py

```diff
@@ -0,0 +1,134 @@
+"""Individual Hessian-Vector-Product and retain_graph
+======================================================
+"""
+# %%
+# imports
+import torch
+from torch import nn
+from torch.autograd import grad
+
+from backpack import backpack, extend
+from backpack.extensions import BatchGrad
+from backpack.utils.examples import load_one_batch_mnist
+
+BATCH_SIZE = 32
+DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+torch.manual_seed(0)
+
+
+# %%
+# Let's define a function that calculates individual hessian_vector products.
+# This function was provided by Haonan Wang (github: haonan3) in #213.
+def batch_hvp(model, loss, params_list, batch_grad_list):
+    if len(params_list) != len(batch_grad_list):
+        raise (ValueError("w and v must have the same length."))
+
+    with backpack(retain_graph=True):
+        one_sample_grad_list = grad(
+            loss, params_list, retain_graph=True, create_graph=True
+        )
+
+    elemwise_products = 0.0
+    for grad_elem, v_elem in zip(one_sample_grad_list, batch_grad_list):
+        sum_over_dims = []
+        for i in range(len(v_elem.shape)):
+            sum_over_dims.append(i)
+        sum_over_dims = tuple(sum_over_dims[1:])
+        elemwise_products += torch.sum(
+            grad_elem.unsqueeze(0) * v_elem.detach(), sum_over_dims
+        )
+
+    with backpack(BatchGrad()):
+        elemwise_products.backward()  # problem: has no attribute 'input0'
+        return_grads = [p.grad_batch for p in model.parameters() if p.requires_grad]
+
+    return return_grads
+
+
+# %%
+# create the model and do one forward pass
+def make_model():
+    return nn.Sequential(
+        nn.Conv2d(1, 10, 5, 1),
+        nn.Sigmoid(),
+        nn.MaxPool2d(2, 2),
+        nn.Conv2d(10, 20, 5, 1),
+        nn.Sigmoid(),
+        nn.MaxPool2d(2, 2),
+        nn.Flatten(),
+        nn.Linear(4 * 4 * 20, 50),
+        nn.Sigmoid(),
+        nn.Linear(50, 10),
+    )
+
+
+model = make_model().to(DEVICE)
+loss_function = torch.nn.CrossEntropyLoss().to(DEVICE)
+model = extend(model)
+loss_function = extend(loss_function)
+
+x, y = load_one_batch_mnist(BATCH_SIZE)
+x, y = x.to(DEVICE), y.to(DEVICE)
+loss = loss_function(model(x), y)
+
+# %%
+# Create a random vector and apply the batch_hvp function.
+params_list = list(model.parameters())
+batch_grad_list = [torch.rand(1, *param.shape, device=DEVICE) for param in params_list]
+
+batch_hvp = batch_hvp(model, loss, params_list, batch_grad_list)
+
+print("\nresult")
+for result in batch_hvp:
+    print(result.shape)
+
+
+# %%
+# Compare the result with the for-loop version
+def batch_hvp_for_loop(model, loss, params_list, batch_grad_list):
```

Comment:
Docstring, with shapes.

Reply:
I suspect you need access to the `input` and `labels` to implement per-sample Hessian-vector products.